spec/rspec/matchers/built_in/yield_spec.rb

```diff
@@ -628,6 +695,15 @@ def each_arg(*args, &block)
         |         got: ["food", "barn"]
       EOS
     end
+
+    it 'fails when the successively yielded args match the matchers at return time only' do
+      expect { |b|
+        %w[ barn food ].each do |val|
+          _yield_with_args(val, &b)
+          val.delete!(val)
```

Comment:
I've not seen `String#delete!` before, and this looks pretty odd.  It takes me a minute to realize what it's doing.  WDYT about using `val.sub!(/.+/, '')` (e.g. replace the contents of the entire string with a blank string) instead?